src/rand/rpo.rs

```diff
@@ -145,8 +145,8 @@ impl RandomCoin for RpoRandomCoin {
         self.state[RATE_START] += nonce;
         Rpo256::apply_permutation(&mut self.state);
 
-        // reset the buffer
-        self.current = RATE_START;
+        // reset the buffer and skip the element containing the proof-of-work
+        self.current = RATE_START + 1;
```

Comment:
This is relevant only for the first invocation of the PRNG, right? On all subsequent invocations do we still need to skip the first element?

But also, I'm not sure I understood how the element containing proof of work gets here in the first place.

Reply:
Could we also explain in the comment *why* we want to skip the element containing the proof-of-work? I'm not familiar with the details of RPO yet so I can imagine it being a bit difficult to understand why we do this when later reviewing that code

Reply:
> This is relevant only for the first invocation of the PRNG, right? On all subsequent invocations do we still need to skip the first element?
> 

This is relevant only to the method which generates random integers and this is the last method used in our setting. Hence this skipping happens only once and only at the final absorption before the final sequence of random integer squeezes. To re-iterate, only the method for drawing integers, which is only relevant to generating FRI queries, is modified in this PR.

> But also, I'm not sure I understood how the element containing proof of work gets here in the first place.

When grinding, the prover will iterate over a sequence of nonces and will apply `check_leading_zeros` to each of these nonces on the current RPO state, which boils down to an invocation of RPO permutation. The nonce will be a valid proof-of-work if the resulting state of the previous step has at least grinding-bits many trailing zeros in its rate's first element. This nonce is called the proof-of-work nonce and is sent to the verifier.
Using the proof-of-work nonce, both prover and verifier will now call `draw_integers` which will absorb the nonce and generate the random integers used for FRI, and given the above, the first element of the resulting state will have at least grinding-bits many trailing zeros. Now, if we set the buffer of random field elements to include all of the state then, we have the possibility that the first generate index will always be zero.
One could probably analyze things further and conclude that it is fine to use this distribution-skewed element, even though this might seem like a bad idea. Without such an analysis, we should just skip using this element.

Reply:
> Could we also explain in the comment _why_ we want to skip the element containing the proof-of-work? I'm not familiar with the details of RPO yet so I can imagine it being a bit difficult to understand why we do this when later reviewing that code

Yep, will do

Reply:
Makes sense! Thank you for the explanation. I would probably expand the comment for this line a bit to contain a subset of what you've written above. Maybe something like this:
```rust
// move the next random element pointer to the second rate element. this is done because
// the first rate element is "biased" via the provided `nonce` to contain some number of
// leading zeros.
```

Reply:
Sure thing. Added!